common/djangoapps/enrollment/tests/test_views.py

```diff
@@ -276,12 +276,88 @@ def test_get_enrollment_with_invalid_key(self):
         self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertIn("No course ", resp.content)
 
-    def _create_enrollment(self, course_id=None, username=None, expected_status=status.HTTP_200_OK, email_opt_in=None, as_server=False):
+    def test_create_enrollment_with_mode(self):
+        """With the right API key, create a new enrollment with a mode set other than the default."""
+        # Create a professional ed course mode.
+        CourseModeFactory.create(
+            course_id=self.course.id,
+            mode_slug='professional',
+            mode_display_name='professional',
+        )
+
+        # Create an enrollment
+        self._create_enrollment(as_server=True, mode='professional')
+
+        self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course.id))
+        course_mode, is_active = CourseEnrollment.enrollment_mode_for_user(self.user, self.course.id)
+        self.assertTrue(is_active)
+        self.assertEqual(course_mode, 'professional')
+
+    def test_update_enrollment_with_mode(self):
```

Comment:
How does the API deal with downgrades (e.g. user is enrolled in verified, API call made to change to honor)?